src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,147 @@
+    {
+        uint32_t relBufferStatus;
+        gss_release_buffer(&relBufferStatus, *outBufferHandle);
+        *outBufferHandle = nullptr;
+        *outBufferLength = 0;
+    }
+    else
+    {
+        *outBufferLength = static_cast<uint32_t>((*outBufferHandle)->length);
+    }
+}
+
+extern "C" uint32_t NetSecurity_AcquireCredSpNego(uint32_t* minorStatus, gss_name_t desiredName, int32_t isInitiate, gss_cred_id_t* outputCredHandle)
+{
+    assert(isInitiate == 0 || isInitiate == 1);
+#if HAVE_GSS_SPNEGO_MECHANISM
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {1, GSS_SPNEGO_MECHANISM};
+#else
+    char gss_mech_value[] = "\x2b\x06\x01\x05\x05\x02";  // Binary representation of SPNEGO Oid (RFC 4178)
+    gss_OID_desc gss_mech_spnego_OID_desc = {6, static_cast<void*>(gss_mech_value)};
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {1, &gss_mech_spnego_OID_desc};
+#endif
+    gss_cred_usage_t credUsage = isInitiate ? GSS_C_INITIATE : GSS_C_ACCEPT;
+    return gss_acquire_cred(minorStatus, desiredName, 0, &gss_mech_spnego_OID_set_desc, credUsage, outputCredHandle, nullptr, nullptr);
+}
+
+extern "C" uint32_t NetSecurity_DeleteSecContext(uint32_t* minorStatus, gss_ctx_id_t* contextHandle)
+{
+    return gss_delete_sec_context(minorStatus, contextHandle, GSS_C_NO_BUFFER);
+}
+
+extern "C" uint32_t NetSecurity_DisplayName(uint32_t* minorStatus, gss_name_t inputName, gss_buffer_t outputNameBuffer)
+{
+    return gss_display_name(minorStatus, inputName, outputNameBuffer, nullptr);
+}
+
+extern "C" uint32_t NetSecurity_DisplayStatus(uint32_t* minorStatus, uint32_t statusValue, int32_t isGssMechCode, gss_buffer_t *outBufferHandle, uint32_t *statusLength)
+{
+    assert (isGssMechCode == 0 || isGssMechCode == 1);
+    int statusType = isGssMechCode ? GSS_C_MECH_CODE : GSS_C_GSS_CODE;
+    *outBufferHandle = reinterpret_cast<gss_buffer_t>(malloc(sizeof(gss_buffer_desc)));
+    uint32_t majorStatus = gss_display_status(minorStatus, statusValue, statusType, GSS_C_NO_OID, nullptr, *outBufferHandle);
+    NetSecurity_HandleError(majorStatus, outBufferHandle, statusLength);
+    return majorStatus;
+}
+
+extern "C" uint32_t NetSecurity_ImportUserName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, gss_name_t* outputName)
+{
+    gss_buffer_desc inputNameBuffer {inputNameLen, inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_C_NT_USER_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t NetSecurity_ImportPrincipalName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, gss_name_t* outputName)
+{
+    gss_buffer_desc inputNameBuffer {inputNameLen, inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_KRB5_NT_PRINCIPAL_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t NetSecurity_InitSecContext(uint32_t* minorStatus, gss_cred_id_t claimantCredHandle, gss_ctx_id_t* contextHandle,
```

Comment:
Please run the code format tool, parameters should either be all on one line or one parameter per line if there are too many